src/libraries/Microsoft.PowerFx.Core/Texl/Intellisense/SuggestionHandlers/FunctionRecordNameSuggestionHandler.cs

```diff
@@ -152,9 +153,17 @@ private static bool TryGetParentRecordFieldType(DType aggregateType, TexlNode cu
             internal static bool AddAggregateSuggestions(DType aggregateType, IntellisenseData.IntellisenseData intellisenseData, int cursorPos)
             {
                 var suggestionsAdded = false;
+                var parentRecordNode = intellisenseData.CurNode.Parent as RecordNode;
+                var alreadyUsedFields = parentRecordNode?.Ids.Select(id => id.Name).ToImmutableHashSet();
```

Comment:
>    var alreadyUsedFields = parentRecordNode?.Ids.Select(id => id.Name).ToImmutableHashSet(); [](http://example.com/codeflow?start=12&length=93)

Consider

                var parentRecordNames = parentRecordNode?.Ids.Select(id => id.Name);
                var alreadyUsedFields = parentRecordNames.ToImmutableHashSet() ?? Enumerable.Empty<DName>().ToImmutableHashSet();


That way, you don't need to check for null in loop below.

Reply:
I feel we should keep the existing method as it would give us a slight perf advantage, it would save us from empty object creation and call to contains on that empty collection.

Reply:
empty object is a static object shared across so it doesn't get created every time. Also in future, you don't need worry about this object being null if you extend this code, so from code maintenance pov, I prefer this way.

Reply:
Good point, I'll update it.